application/src/test/java/org/togetherjava/tjbot/commands/tophelper/TopHelperMessageListenerTest.java

```diff
@@ -0,0 +1,194 @@
+package org.togetherjava.tjbot.commands.tophelper;
+
+import net.dv8tion.jda.api.entities.Message;
+import net.dv8tion.jda.api.entities.User;
+import net.dv8tion.jda.api.entities.channel.ChannelType;
+import net.dv8tion.jda.api.events.message.MessageReceivedEvent;
+import net.dv8tion.jda.internal.JDAImpl;
+import net.dv8tion.jda.internal.entities.UserImpl;
+import net.dv8tion.jda.internal.entities.channel.concrete.ThreadChannelImpl;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import org.togetherjava.tjbot.config.Config;
+import org.togetherjava.tjbot.config.HelpSystemConfig;
+import org.togetherjava.tjbot.db.Database;
+import org.togetherjava.tjbot.jda.JdaTester;
+
+import java.util.List;
+import java.util.stream.Stream;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.Mockito.*;
+import static org.togetherjava.tjbot.db.generated.tables.HelpChannelMessages.HELP_CHANNEL_MESSAGES;
+
+final class TopHelperMessageListenerTest {
+
+    private static final String STAGING_CHANNEL_PATTERN = "ask_here";
+    private static final String OVERVIEW_CHANNEL_PATTERN = "active_questions";
+
+    private static JdaTester jdaTester;
+    private static TopHelpersMessageListener topHelpersListenerSpy;
+
+    @BeforeAll
+    static void setUp() {
+        Database database = Database.createMemoryDatabase(HELP_CHANNEL_MESSAGES);
+        Config config = mock(Config.class);
+        HelpSystemConfig helpSystemConfig = mock(HelpSystemConfig.class);
+
+        when(helpSystemConfig.getStagingChannelPattern()).thenReturn(STAGING_CHANNEL_PATTERN);
+        when(helpSystemConfig.getOverviewChannelPattern()).thenReturn(OVERVIEW_CHANNEL_PATTERN);
+
+        when(config.getHelpSystem()).thenReturn(helpSystemConfig);
+
+        jdaTester = new JdaTester();
+        topHelpersListenerSpy = spy(new TopHelpersMessageListener(database, config));
+    }
+
+    @Test
+    @DisplayName("Recognizes valid messages")
+    void recognizesValidMessages() {
+        // GIVEN a message by a human in a help channel
+        MessageReceivedEvent event =
+                createFakeMessageRecievedEvent(false, false, true, OVERVIEW_CHANNEL_PATTERN);
+
+        // WHEN checking if the message should be ignored
+        boolean shouldBeIgnored = topHelpersListenerSpy.shouldIgnoreMessage(event);
+
+        // THEN the message is not ignored
+        assertFalse(shouldBeIgnored);
+    }
+
+    @Test
+    @DisplayName("Ignores bots")
```

Comment:
ur display names are obsolete since they dont add anything extra on the method name. remove them or write a more detailed description (same for others)

Reply:
Ah ok. I did not know the tests get their names automagically. Never wrote a test before xD